### DISCUSSION TITLE
brew upgrade delete my XAMPP folder at Applications on macOS

### DISCUSSION THREAD

**Discussion by Rudolf247** [Homebrew/discussions#4278] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: ba6d87eed9ce31f42fa2b33a4e6067d9cf17f13f
Last commit: 8 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 0b7a00899d0a5eaf44370a940ce326591fce11e3
Core tap last commit: 5 days ago
Core tap branch: master
Core tap JSON: 23 Feb 12:30 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.39.2 => /opt/homebrew/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I used 'brew upgrade' and in the process deleted my XAMPP folder where I had very important files. Of course most of them were on Git, but not all of them. I know this shouldn't happen, but I was looking in Terminal as the upgrade process was being performed and suddenly there was a deleting and before I could stop it, the folder was gone.

**Comment by SMillerDev** [comment]
If it's the app directory, yes. That's what the updates do, remove the old package and install a new one.

**Reply by Rudolf247** [reply]
Thank you for your reply. I'm a new MacBook user so I didn't know this, anyway why does XAMPP create a htdocs folder in a folder it deletes when updating? That's not very well thought out. So I lost the files? No way to get them back?
Thank you

**Reply by gromgit** [reply]
> why does XAMPP create a htdocs folder in a folder it deletes when updating?

That's a question for the XAMPP developers. I first thought that this was a result of the uninstall stanza in the `xampp` cask:
```
  uninstall quit:   "com.bitnami.manager",
            script: {
              executable: "/Applications/XAMPP/uninstall.app/Contents/MacOS/osx-x86_64",
              args:       ["--mode", "unattended"],
              sudo:       true,
            },
            delete: "/Applications/XAMPP/"
```
so I remove the `delete:` key and did a full install/reinstall cycle, then checked the results:
```
% ls -lR /Applications/XAMPP 
total 0
drwxrwxr-x  4 root  admin  128 Mar  1 12:38 xamppfiles

/Applications/XAMPP/xamppfiles:
total 0
drwxrwxr-x  2 root    admin   64 Mar  1 12:38 etc
drwxrwxrwx  3 daemon  daemon  96 Mar  1 12:37 temp

/Applications/XAMPP/xamppfiles/etc:
total 0

/Applications/XAMPP/xamppfiles/temp:
total 0
drwxr-xr-x  2 _mysql  _mysql  64 Mar  1 12:37 mysql

/Applications/XAMPP/xamppfiles/temp/mysql:
total 0
```
So yeah, it's the XAMPP uninstaller that's nuking all your files. Homebrew merely cleans up the orphaned bits left over.

> So I lost the files? No way to get them back?

Probably not, unless you'd previously set up a Time Machine backup. If you hadn't, you might want to do so soonest: https://support.apple.com/en-us/HT201250
